test/core/rtsmith_api_test.cpp

```diff
@@ -104,6 +106,93 @@ TEST_F(P4RuntimeApiTest, GeneratesATestViaTheApi) {
     ASSERT_TRUE(rtSmithResultOpt.has_value());
 }
 
+// Tests for the optimization of various expressions with user-specified fuzzer configurations (by
+// ingesting a TOML file).
+TEST_F(P4RuntimeApiTest, GeneratesATestViaTheApiWithFuzzerConfig) {
+    auto source = generateTestProgram(R"(
+    // Drop the packet.
+    action acl_drop() {
+        mark_to_drop(sm);
+    }
+
+    table drop_table {
+        key = {
+            hdr.eth_hdr.dst_addr : ternary @name("dst_eth");
+        }
+        actions = {
+            acl_drop();
+            @defaultonly NoAction();
+        }
+    }
+
+    apply {
+        if (hdr.eth_hdr.isValid()) {
+            drop_table.apply();
+        }
+    })");
+    auto autoContext = SetUp("bmv2", "v1model");
+    auto &rtSmithOptions = RTSmith::RtSmithOptions::get();
+    rtSmithOptions.target = "bmv2"_cs;
+    rtSmithOptions.arch = "v1model"_cs;
+    const char *configFilePath = "configuration.toml";
```

Comment:
Inline this configuration. Or make it part of the test folder instead.

Reply:
Sometimes it is useful to have an API that also takes in a string (over a file). Precisely for directly ingesting a string.

Reply:
Another approach is to define a config. Write it to a temporary folder, then ingest it again.

Reply:
To test the configuration parsing more exhaustively, I've implemented both (i.e., parsing the configurations specified in the string representation and parsing the configurations specified in the file). I added an option to input/ingest the configurations in a string (via the added flag `--toml-string`).